addons/supplies/XEH_preInit.sqf

```diff
@@ -0,0 +1,9 @@
+#include "script_component.hpp"
+
+ADDON = false;
+
+#include "XEH_PREP.hpp"
+
+[QGVAR(Medic), "init", {(_this select 0) call FUNC(medicalArsenal)}, false, [], true] call CBA_fnc_addClassEventHandler;
```

Comment:
```suggestion
[QGVAR(Medic), "init", LINKFUNC(medicalArsenal)] call CBA_fnc_addClassEventHandler;
```
It's `preInit`, so retroactive running is not necessary. There is also no reason to not do that on inherited boxes.